changelog.d/gh-7403.fixed

```diff
@@ -0,0 +1,4 @@
+Matching: Fixed a bug where explicit casts of expressions would produce two matches to
```

Comment:
I think it's simpler than this, any expression pattern `E` matches `E` as well as `(T) E`. And e.g. `x` matches `x` and `(int) x`.